lib/manager/dockerfile/extract.ts

```diff
@@ -6,12 +6,27 @@ import * as ubuntuVersioning from '../../versioning/ubuntu';
 import type { PackageDependency, PackageFile } from '../types';
 
 export function splitImageParts(currentFrom: string): PackageDependency {
-  if (currentFrom.includes('$')) {
-    return {
-      skipReason: SkipReason.ContainsVariable,
-    };
+  // Check if we have a variable in format of "${VARIABLE:-<image>:<defaultVal>@<digest>}"
+  // If so, remove everything except the image, defaultVal and digest.
+  let isVariable = false;
+  let hasDefaultValue = false;
+  let cleanedCurrentFrom: string = currentFrom;
+  if (currentFrom.startsWith('${') && currentFrom.endsWith('}')) {
+    isVariable = true;
+
+    if (
+      currentFrom.match(/\$/g).length === 1 && // Ensure it has exactly one '$' to avoid the cases we don't support
```

Comment:
I think we should use a re2 regex here. Should be easier to validate variable with default value.

The regex should also be a module const.

Reply:
Hi @viceice I tried to use re2, but ran into some strange problems, probably because of lack of my experience with re2. So, I decided to replace RegEx with a usual `String.split()`, which seems to be more simple.